### ISSUE TITLE
ConcurrentModificationException logged in file tree observer

### ISSUE THREAD

**Issue by pjfanning** [sbt/sbt#6188]
## steps

* SBT 1.4.4, Scala 2.13.4, Zulu JDK 8.0.252, MacOs

## problem

I've recently seen this a few times. Seems not to cause issues.

```
java.util.ConcurrentModificationException
  | => nat java.base/java.util.LinkedHashMap$LinkedHashIterator.nextNode(LinkedHashMap.java:719)
	at java.base/java.util.LinkedHashMap$LinkedEntryIterator.next(LinkedHashMap.java:751)
	at java.base/java.util.LinkedHashMap$LinkedEntryIterator.next(LinkedHashMap.java:749)
	at scala.collection.convert.Wrappers$JMapWrapperLike$$anon$5.next(Wrappers.scala:288)
	at scala.collection.convert.Wrappers$JMapWrapperLike$$anon$5.next(Wrappers.scala:285)
	at scala.collection.MapLike$$anon$2.next(MapLike.scala:223)
	at scala.collection.Iterator.foreach(Iterator.scala:943)
	at scala.collection.Iterator.foreach$(Iterator.scala:943)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
	at scala.collection.MapLike$DefaultValuesIterable.foreach(MapLike.scala:213)
	at sbt.internal.nio.Observers.onNext(Observers.scala:117)
	at sbt.internal.nio.FileTreeRepositoryImpl$$anon$1.$anonfun$onDelete$1(FileTreeRepositoryImpl.scala:63)
	at sbt.internal.nio.FileTreeRepositoryImpl$$anon$1.$anonfun$onDelete$1$adapted(FileTreeRepositoryImpl.scala:62)
	at scala.util.Either.foreach(Either.scala:270)
	at sbt.internal.nio.FileTreeRepositoryImpl$$anon$1.onDelete(FileTreeRepositoryImpl.scala:62)
	at com.swoval.files.CacheObservers.onDelete(CacheObservers.java:43)
	at com.swoval.files.FileCacheDirectoryTree$6.run(FileCacheDirectoryTree.java:512)
	at com.swoval.files.FileCacheDirectoryTree$4.run(FileCacheDirectoryTree.java:258)
	at com.swoval.files.Executor$PriorityRunnable.run(Executor.java:161)
	at com.swoval.files.Executor$ExecutorImpl$1.run(Executor.java:65)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

## expectation

Ideally, no errors should be logged.

**Comment by eatkins** [comment]
I think this should be fixed in 1.4.5. There was a PR to address it in the io repository https://github.com/sbt/io/pull/310.

**Comment by eed3si9n** [comment]
I'm going to close this as fixed then.
